### ISSUE TITLE
Impossible magic bolt training

### ISSUE THREAD

**Issue by Pegwald** [aura-project/aura#73]
I am currently checking out how the magic bolts are working and I've figured that the requirement for "knocking down an enemy using a combo attack" is not really giving any training exp. I can spam 5 ice bolts or knock down an enemy continuously with lightning bolt without getting anything for it. Just a dead mob. 
![bolts](https://cloud.githubusercontent.com/assets/13089450/10988024/e886f6a0-8438-11e5-9efe-111780f3b929.PNG)

**Comment by exectails** [comment]
The check right now is for knock down on an alive mob, so you'd have to knock it down without killing it. Is that working? Should the training method be fulfilled with killing as well?

**Comment by QewQew** [comment]
>The check right now is for knock down on an alive mob

But there's also a check for Death that applies the KnockDown flag that is checked in Train, so shouldn't it provide training even for knockdown + death?
https://github.com/aura-project/aura/blob/master/src/ChannelServer/Skills/Base/MagicBolt.cs#L218

**Comment by exectails** [comment]
KnockDown and FinishingKnockDown are two different things. Our naming scheme is a little unfortunate^^"

`FinishingKnockDown = Finished | KnockDownFinish | FinishingHit`

**Comment by Pegwald** [comment]
yep, I've spammed rF icebolt on a poor defenseless little goblin. It was a long and painful death, unfortunately. R.I.P.

**Comment by exectails** [comment]
I've just tried to knock down a bear, I wasn't able to. So... is the problem that you can't knock *down* enemies, or is it that you don't get training for it?

**Comment by Pegwald** [comment]
The issue was that you didn't get the training exp for knocking them down. You gain the exp when you knock down the enemy while it's being pushed back, I think. This used to be tricky in NA with high ping, but possible with the use of a pet. That's all I know. I am going to log in and check it out if it's fixed.

**Comment by Pegwald** [comment]
It is working for ice bolt, but the mob will not be knocked down by lightning bolt. It just keeps being knocked back until the balance resets and the mob comes flying back at you immediately. If you try charging two lightning bolts and fire it at a single mob, it also just gets knocked back. That is very strange, because in NA they would get knocked down instantly if you charged 2+ lightning bolts on a single mob, 3+ charges for 2 mobs and so on.
My conclusion is therefore that the lightning bolt is lacking a knockdown ability. Do I make a new issue for this request?

**Comment by exectails** [comment]
No, that's fine, it's still the same issue, and all bolts use the same base class.

Lightning Bolt follows the same rules as Icebolt, if you get the stability down, it gets knocked back, if you hit it again while the stability is down, it gets knocked down, and only after that, if it's hit while being down, it should reset the stun.

I did miss that knock back rule for LB, that will have to be fixed. The question is if afterwards you'll be able to knock down, by knocking back first, and then casting another LB. Is that how it works on NA?

**Comment by exectails** [comment]
Give this a try. It should work now, the question is if it feels right.

**Comment by Pegwald** [comment]
It works correctly now. Good job!
